File: backend/app/schemas/data_management.py
from typing import Literal

from pydantic import BaseModel


DataResetScope = Literal["transactions_only", "transactions_and_snapshots"]


class DataResetRequest(BaseModel):
    scope: DataResetScope


class DataResetCounts(BaseModel):
    transactions: int
    asset_snapshots: int
    investments: int
    loans: int


class DataResetResponse(BaseModel):
    scope: DataResetScope
    deleted: DataResetCounts
    upload_logs_retained: bool = True

File: frontend/src/api/dataManagement.ts
import { apiRequest, getApiKeyHeaders } from './client';

export type DataResetScope = 'transactions_only' | 'transactions_and_snapshots';

export interface DataResetResponse {
  scope: DataResetScope;
  deleted: {
    transactions: number;
    asset_snapshots: number;
    investments: number;
    loans: number;
  };
  upload_logs_retained: boolean;
}

export function resetData(scope: DataResetScope) {
  return apiRequest<DataResetResponse>('/data/reset', {
    method: 'POST',
    body: { scope },
    headers: getApiKeyHeaders(),
  });
}
